azkaban-common/src/main/java/azkaban/executor/ExecutorApiGateway.java

```diff
@@ -116,7 +122,7 @@ String createExecutionPath(int execId) {
 
       // Ideally we should throw an exception if executionId is null but some existing code
       // (updateExecutions()) expects to call this method with a null executionId.
-      String executionPath = (executionId == null) ? null : createExecutionPath(executionId);
+      String executionPath = createExecutionPath(Optional.ofNullable(executionId));
```

Comment:
I hope it's ok to exit with an unhandled (and unreported, at least not yet) exception here if `createExecutionPath` throws.

Reply:
It should be ok as `callWithExecutionId` already throws that exception, however for good measure I have updated `createExecutionPath` to log the exception. 
Please let me know if I misunderstood your comment.